client/src/components/bash-button.tsx

```diff
@@ -0,0 +1,149 @@
+*/
+import { Button, Typography } from "@material-ui/core";
+import React from "react";
+import { navigate } from "gatsby";
+import PropTypes from "prop-types";
+import { useWithReviewAnswerState } from "hooks/graphql/use-with-review-answer-state";
+import { Answer, MentorType, Status, UtteranceName } from "types";
+
+export default function BashButton(props: {
+  accessToken: string;
+  setThumbnail: (file: File) => void;
+}): JSX.Element {
+  const { mentor } = useWithReviewAnswerState(props.accessToken, {
+    subject: undefined,
+  });
+  const bash = {
+    text: "",
+    reason: "",
+    action: () => {
+      undefined;
+    },
+  };
+  const idle = mentor?.answers.find(
+    (a) => a.question.name === UtteranceName.IDLE
+  );
+  const categories: {
+    subjectName: string;
+    subject: string;
+    categoryName: string;
+    category: string | undefined;
+    answers: Answer[];
+  }[] = [];
+
+  mentor?.subjects.forEach((s) => {
+    categories.push({
+      subjectName: s.name,
+      subject: s._id,
+      categoryName: "Uncategorized",
+      category: undefined,
+      answers: mentor?.answers.filter((a) =>
+        s.questions
+          .filter((q) => !q.category)
+          .map((q) => q.question._id)
+          .includes(a.question._id)
+      ),
+    });
+    s.categories.forEach((c) => {
+      categories.push({
+        subjectName: s.name,
+        subject: s._id,
+        categoryName: c.name,
+        category: c.id,
+        answers: mentor?.answers.filter((a) =>
+          s.questions
+            .filter((q) => q.category?.id === c.id)
+            .map((q) => q.question._id)
+            .includes(a.question._id)
+        ),
+      });
+    });
+  });
+  const firstIncomplete = categories.find((c) =>
+    c.answers.find((a) => a.status === Status.INCOMPLETE)
+  );
+
+  switch (true) {
+    case mentor?.thumbnail == "":
+      bash.text = "Add a Thumbnail";
+      bash.reason = "A thumbnail helps a user identify your mentor";
+      bash.action = () => {
+        undefined;
+      };
+      break;
+    case idle?.status === Status.INCOMPLETE &&
+      mentor?.mentorType === MentorType.VIDEO:
+      bash.text = "Record an Idle Video";
+      bash.reason = "Users see your idle video while typing a question";
+      bash.action = () => {
+        navigate(
+          `/record?back=${encodeURI(`/?subject=${undefined}`)}&videoId=${
+            idle?.question._id
+          }`
+        );
+      };
+      break;
+    case Boolean(firstIncomplete):
+      bash.text = "Answer " + firstIncomplete?.categoryName + " Questions";
+      bash.reason =
+        "You have unanswered questions in the " +
+        firstIncomplete?.subjectName +
+        " subject";
+      bash.action = () =>
+        navigate(
+          `/record?back=${encodeURI(
```

Comment:
use `URLSearchParams` to construct/encode query strings.

Maybe easier if we had a global helper function , e.g.

```typescript
function urlBuild(base: string, params: Record<string, string>) {
  const query = URLSearchParams()
  Object.keys(params).forEach(n => query.append(n, params[n])
  return `${base}?{query.toString()}`
}
```

Reply:
I think this is about the only thing I'm missing. Should we use this function everywhere we're generating urls?